lib/Runtime/Library/JavascriptBigInt.h

```diff
@@ -4,44 +4,46 @@
 //-------------------------------------------------------------------------------------------------------
 #pragma once
 
+
 namespace Js
 {
     class JavascriptBigInt sealed : public RecyclableObject
     {
-
     private:
-        Field(uint32*) m_digits;         // digits
-        Field(uint32) m_length;          // length
+        Field(digit_t*) m_digits;         // digits
+        Field(digit_t) m_length;          // length
+        Field(digit_t) m_maxLength;          // max length without resize
         Field(bool) m_isNegative;
 
         // TODO: BigInt should be arbitrary-precision, need to resize space and add tagged BigInt
-        static const uint32 MaxStringLength = 100; // Max String length
-        static const uint32 MaxDigitLength = 20;  // Max Digit length 
-        Field(uint32) m_reservedDigitsInit[MaxDigitLength]; // pre-defined space to store array
+        static const digit_t InitDigitLength = 2;  // Max Digit length 
 
         DEFINE_VTABLE_CTOR(JavascriptBigInt, RecyclableObject);
 
     public:
         JavascriptBigInt(StaticType * type)
-            : RecyclableObject(type), m_length(0), m_digits(nullptr), m_isNegative(false)
+            : RecyclableObject(type), m_length(0), m_digits(nullptr), m_isNegative(false), m_maxLength(InitDigitLength)
         {
             Assert(type->GetTypeId() == TypeIds_BigInt);
         }
 
-        JavascriptBigInt(const char16 * content, charcount_t cchUseLength, bool isNegative, StaticType * type)
+        JavascriptBigInt(const char16 * content, charcount_t cchUseLength, bool isNegative, ScriptContext * scriptContext, StaticType * type)
             : JavascriptBigInt(type)
         {
             Assert(type->GetTypeId() == TypeIds_BigInt);
-            InitFromCharDigits(content, cchUseLength, isNegative);
+            InitFromCharDigits(content, cchUseLength, isNegative, scriptContext);
         }
 
 
         static bool LessThan(Var aLeft, Var aRight);
         static bool Equals(Var aLeft, Var aRight);
+        static Var Increment(Var aRight, ScriptContext * scriptContext);
+        static Var Decrement(Var aRight, ScriptContext * scriptContext);
 
         inline BOOL isNegative() { return m_isNegative; }
 
         static JavascriptBigInt * Create(const char16 * content, charcount_t cchUseLength, bool isNegative, ScriptContext * scriptContext);
+        static JavascriptBigInt * CloneToScriptContext(Var aValue, ScriptContext* scriptContext);
```

Comment:
Could you please change this signature to `RecyclableObject * CloneToScriptContext(ScriptContext* requestContext) override` so it overrides the method inherited from RecyclableObject? There aren't any tests for that case yet, but we'll need it eventually.

Reply:
will do